Orange/widgets/data/oweditdomain.py

```diff
@@ -1555,6 +1565,18 @@ def get_data(self):
             tr.insert(0, self.format_cb.currentData())
         return var, tr
 
+    def _on_custom_change(self):
+        self.custom_text = self.custom_edit.text()
+        date_pat = r"%(-?)d|%(b|B)|%(-?)m|%(y|Y)|%(-?)j|%(-?)U|%(-?)W|%(a|A)|%w"
+        time_pat = r"%(-?)H|%(-?)I|%p|%(-?)M|%(-?)S|%f"
+        pattern1 = re.compile(date_pat)
+        pattern2 = re.compile(time_pat)
+        self.have_date = int(bool(pattern1.search(self.custom_text)))
+        self.have_time = int(bool(pattern2.search(self.custom_text)))
+        if self.format_cb.currentText() != "Custom format":
```

Comment:
Define a constant `CUSTOM_FORMAT = "Custom format"` and use it when adding the item to combo and here. Otherwise, one can change the combo label and forget to change it here (and elsewhere, as necessary).